lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -259,6 +259,131 @@ class ExportCommand < HammerCLIKatello::SingleResourceCommand
       end
     end
 
+    class ExportCommand < HammerCLIKatello::SingleResourceCommand
+      include HammerCLIForemanTasks::Async
+
+      action :export
+      command_name "export"
+
+      success_message _("Content view export is available in %{directory}.")
+      failure_message _("Could not export the content view")
+
+      option "--id", "ID", _("Content View Version numeric identifier")
+      option(
+        '--export-dir',
+        'EXPORT_DIR',
+        _("Directory to put content view version export into.")
+      )
+
+      validate_options do
+        option(:option_export_dir).required
+      end
+
+      build_options
+
+      class ContentViewIdParamSource
+        def initialize(command)
+          @command = command
+        end
+
+        def get_options(_defined_options, result)
+          if result['option_id'].nil? && result['option_name']
+            result['option_id'] = @command.resolver.content_view_id(result)
+          end
+          result
+        end
+      end
+
+      def option_sources
+        sources = super
+        idx = sources.index { |s| s.class == HammerCLIForeman::OptionSources::IdParams }
+        sources.insert(idx, ContentViewIdParamSource.new(self))
+        sources
+      end
+
+      def execute
+        cvv = content_view_version
+        repositories = cvv['repositories'].collect do |repo|
+          repository(repo['id'])
+        end
+
+        repositories.each do |repo|
+          repo['packages'] = packages(repo['id'])
+          repo['errata'] = errata(repo['id'])
+        end
+
+        json = export_json(cvv, repositories)
+        create_tar(cvv, repositories, json)
+      end
+
+      def create_tar(cvv, repositories, json)
+        export_file = "export-#{cvv['id']}.json"
+        export_repos_tar = "exported-repos-#{cvv['id']}.tar"
+        export_tar = "export-#{cvv['id']}.tar"
+        published_repo_dir = "/var/lib/pulp/published/yum/https/repos/"
```

Comment:
Let's make this a constant. You're using it as a literal in 3 places.

Reply:
Given this is only used within the context of this function, shouldn't I prefer to keep it scoped?

Reply:
Not sure I understand your concern?

Reply:
I'm suggesting you put it in `ExportCommand`:

```rb
class ExportCommand
  PUBLISHED_REPO_DIR = /var/lib/pulp/published/yum/https/repos/
  ...
end
```

In your tests, you can access the constant with `HammerCLIKatello::ContentViewVersion::ExportCommand::REPO_DIR`

Reply:
I'm OK with explicitness in the tests given this is used only in this local method. If you require it changed, I will.

Reply:
> this is used only in this local method

But it's also used in [`export_json`](https://github.com/Katello/hammer-cli-katello/pull/589/files#diff-c16d36ded485c9e83859c586343e3bb6R377) (as I said, you're using it in 2 places, not including tests). Additionally, it's more explicit IMO to use a named variable/constant than a literal. It's not blocking, but I just don't understand your pushback (I did ask for clarification but haven't received a response).